src/pages/Flow/Actions.vue

```diff
@@ -55,8 +56,47 @@ export default {
         return result
       }, true)
     },
-    isReadOnlyUser() {
-      return this.role === 'READ_ONLY_USER'
+    disableToggle() {
+      let isDisabled = false
+      if (
+        this.hasPermission('create', 'run') &&
+        this.hasPermission('delete', 'run')
+      ) {
+        isDisabled = false
+      }
+      if (
+        !this.hasPermission('create', 'run') &&
+        !this.hasPermission('delete', 'run')
+      ) {
+        isDisabled = true
+      }
+      if (
+        this.hasPermission('create', 'run') &&
+        !this.hasPermission('delete', 'run') &&
+        !this.isScheduled
+      ) {
+        isDisabled = false
+      } else if (
+        this.hasPermission('create', 'run') &&
+        !this.hasPermission('delete', 'run') &&
+        this.isScheduled
+      ) {
+        isDisabled = true
+      }
+      if (
+        this.hasPermission('delete', 'run') &&
+        !this.hasPermission('create', 'run') &&
+        this.isScheduled
+      ) {
+        isDisabled = false
+      } else if (
+        this.hasPermission('delete', 'run') &&
+        !this.hasPermission('create', 'run') &&
+        !this.isScheduled
+      ) {
+        isDisabled = true
+      }
+      return isDisabled
```

Comment:
I think this logic can be shortened for readability/maintainability, my suggestion is something like this:
```js
disableToggle() {
  const c = this.hasPermission('create', 'run')
  const d = this.hasPermission('delete', 'run')
  const scheduled = this.isScheduled

  // disable the schedule toggle unless:
  // - a user has both create and delete permissions
  // - the schedule is off and a user has create permissions
  // - the schedule is on and a user has delete permissions
  return c && d || !scheduled && c || scheduled && d
}
```

Reply:
Yup makes more sense. Will also update in the ScheduleToggle component as well.